### PR TITLE
Updates for STU3

### PR DESCRIPTION
**Pull request by xmlmodeling**
Three commits with updates and a bug fix for STU3

### CODE REVIEW

File: App/DataType+Utilities.swift [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    }
+    
+}
+
+
+extension Period {
+    
+    func displayString() -> String? {
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "MMM dd, yyyy"
```
Comment (p2) [inline App/DataType+Utilities.swift]: Please just set `dateFormatter.dateStyle` and `dateFormatter.timeStyle`. This way the format will be in the correct locale for the user.

File: App/ResourceListViewController.swift [hunk 2]
```diff
@@ -90,9 +145,27 @@ class ResourceListViewController: UITableViewController {
...
+            if let displayName: String = goalInstance.description_fhir?.displayString() {
+                cell.textLabel?.text = displayName
+            } else {
+                cell.textLabel?.text = "Goal"
+            }
+            
+            // TODO: in STU3 final, update to use embedded goal target value (extension in v1.8.0)
+            if let goalDate: FHIRDate = goalInstance.targetDate {
+                let dateFormatter = DateFormatter()
+                dateFormatter.dateFormat = "MMM dd, yyyy"
```
Comment (p2) [inline App/ResourceListViewController.swift]: Same as above, just set `dateStyle` and `timeStyle`.

File: App/ResourceListViewController.swift [hunk 3]
```diff
@@ -156,7 +229,19 @@ class ResourceListViewController: UITableViewController {
...
+            
+        else if let referralRequestInstance = resource as? ReferralRequest {
+            if let displayName: String = referralRequestInstance.reason?.displayString() {
+                cell.textLabel?.text = displayName
+            } else {
+                cell.textLabel?.text = "Referral Request"
+            }
+            
+            if let specialty: String = referralRequestInstance.specialty?.displayString() {
+                cell.detailTextLabel?.text = "For specialty: " + specialty
```
Comment (p2) [inline App/ResourceListViewController.swift]: I prefer using `"For specialty: \(specialty)"` but that's just personal preference. :)

File: Endpoints.swift [hunk 4]
```diff
@@ -12,32 +12,39 @@ import SMART
 
 func configuredEndpoints() -> [Endpoint] {
 	var endpoints = [Endpoint]()
-	
-	// Grahame's test server
-	let grahame = Client(
```
Comment (p2) [inline Endpoints.swift]: Keep Grahame's server in the list – soon everything is STU-3.

File: Endpoints.swift [hunk 5]
```diff
@@ -12,32 +12,39 @@ import SMART
...
-			"redirect": "smartapp://callback",
-			"logo_uri": "https://avatars1.githubusercontent.com/u/7401080",
-			])
-	grahame.authProperties.granularity = .patientSelectNative
-	grahame.authProperties.embedded = true
-	endpoints.append(Endpoint(client: grahame, name: "Health Intersections"))
-	
-	// SMART DSTU-2!! sandbox
-	// Credentials obtained by registering on the SMART website
-	let smart = Client(
```
Comment (p2) [inline Endpoints.swift]: Also keep SMART – we'll soon have an STU-3 version and will only need to update the URL. It's fine if not all servers work out of the box, and it's good to have the stubs in code already.

**Review by p2 — COMMENTED** [review]
Thanks Dave! A few small comments, and a big one: please use tabs for indentation to be consistent with the existing codebase. 👍🏼

**Comment by xmlmodeling** [comment]
Update as per Pascal's comments.

**Pull request merged**